File: gateway/src/main/java/com/beat/gateway/security/internal/servlet/MemberAuthentication.java
package com.beat.gateway.security.internal.servlet;

import java.util.Collection;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;

public class MemberAuthentication extends UsernamePasswordAuthenticationToken {

	public MemberAuthentication(
		Object principal,
		Object credentials,
		Collection<? extends GrantedAuthority> authorities
	) {
		super(principal, credentials, authorities);
	}
}

File: gateway/src/main/kotlin/com/beat/gateway/authentication/internal/MemberAuthentication.kt
package com.beat.gateway.authentication.internal

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.GrantedAuthority

/**
 * `ROLE_MEMBER` 인증 토큰. principal은 항상 사용자 memberId다.
 */
class MemberAuthentication(
    memberId: Long,
    authorities: Collection<GrantedAuthority>,
) : UsernamePasswordAuthenticationToken(memberId, null, authorities)
